### PR TITLE
refactor: use `std::uint8_t` for enum

### PR DESCRIPTION
**Pull request by tony-go**
First advice taken from the great tidy!

### CODE REVIEW

File: src/jsonschema/compile_evaluate.cc [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#include <cstdint>
```
Comment (jviotti) [inline src/jsonschema/compile_evaluate.cc:1]: Can you move this to the standard library includes within line 8 to line 17?

File: src/jsonschema/include/sourcemeta/jsontoolkit/jsonschema_anchor.h [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 #include "jsonschema_export.h"
 
+#include <cstdint>
```
Comment (jviotti) [inline src/jsonschema/include/sourcemeta/jsontoolkit/jsonschema_anchor.h:6]: Also here, can you move this below to the std includes section?

File: src/jsonschema/compile_evaluate.cc (hunk 1 above)
Comment (tony-go) [reply src/jsonschema/compile_evaluate.cc:1]: HAHAHAH the formater added them automatically!

**Comment by tony-go** [comment]
Let me re-add headers properly

File: src/jsonschema/compile_evaluate.cc [hunk 3]
```diff
@@ -6,6 +6,7 @@
 
 #include <algorithm>   // std::min, std::any_of
 #include <cassert>     // assert
+#include <cstdint>     // std::uint64_t
```
Comment (jviotti) [inline src/jsonschema/compile_evaluate.cc:9]:
```suggestion
#include <cstdint>     // std::uint8_t
```

?

**Review by jviotti — APPROVED** [review]
I made a tiny suggestion but other than that it looks good!

File: src/jsonschema/compile_evaluate.cc (hunk 3 above)
Comment (tony-go) [reply src/jsonschema/compile_evaluate.cc]: dear focus!

**Pull request merged**